homeassistant/components/spokestack/tts.py

```diff
@@ -0,0 +1,82 @@
+"""Support for the Spokestack Text to Speech Service."""
+import logging
+
+from spokestack.tts.clients.spokestack import TextToSpeechClient, TTSError
+
+from homeassistant.components.tts import CONF_LANG, Provider
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    CONF_IDENTITY,
+    CONF_SECRET_KEY,
+    DEFAULT_IDENTITY,
+    DEFAULT_LANG,
+    DEFAULT_MODE,
+    DEFAULT_PROFILE,
+    DEFAULT_SECRET_KEY,
+    DEFAULT_VOICE,
+    SUPPORTED_LANGUAGES,
+    SUPPORTED_MODES,
+    SUPPORTED_OPTIONS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_engine(hass: HomeAssistant, config, discovery_info=None):
+    """Set up Spokestack TTS provider."""
+    return SpokestackProvider(hass, config)
+
+
+class SpokestackProvider(Provider):
+    """Spokestack TTS Provider."""
+
+    def __init__(self, hass: HomeAssistant, config):
+        """Initialize Spokestack client."""
+        self._client = TextToSpeechClient(
+            key_id=config.get(CONF_IDENTITY, DEFAULT_IDENTITY),
+            key_secret=config.get(CONF_SECRET_KEY, DEFAULT_SECRET_KEY),
```

Comment:
These keys are always there because you require them in the config flow.
```suggestion
            key_id=config[CONF_IDENTITY],
            key_secret=config[CONF_SECRET_KEY],
```